src/kv/kv.h

```diff
@@ -1142,18 +1146,22 @@ namespace kv
         {
           if (curr_map->is_replicated())
           {
+            replicated = true;
             curr_map->serialise(replicated_serialiser, include_reads);
           }
           else
           {
+            derived = true;
             curr_map->serialise(derived_serialiser, include_reads);
           }
         }
       }
 
       // Return serialised Tx.
-      return {std::move(replicated_serialiser.get_raw_data()),
-              std::move(derived_serialiser.get_raw_data())};
+      return {replicated ? std::move(replicated_serialiser.get_raw_data()) :
+                           std::move(std::vector<uint8_t>(0)),
```

Comment:
{}

Reply:
afaik you can't use braced initializer list on a ternary operator

Reply:
Indeed, I didn’t know that!